### PR TITLE
docs(nx-plugin): add documentation for generators/schematics

### PR DESCRIPTION
**Pull request by luishcastroc**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [ ] nx-plugin
- [ ] trpc

## What is the current behavior?

No documentation for generators/schematics

## What is the new behavior?

New tab created for schematic and Nx documentation updated.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by luishcastroc** [comment]
i took the opportunity to try to test the schematic part (which was pending) and it seems that is not generating the page in the proper place, i'll open an issue on that, i don't think we even need a project for that to be generated properly

**Comment by luishcastroc** [comment]
> i took the opportunity to try to test the schematic part (which was pending) and it seems that is not generating the page in the proper place, i'll open an issue on that, i don't think we even need a project for that to be generated properly

Seems that for schematics the project needs to be / for the process to work properly, for Nx if the project is empty it takes the default project but it doesn't work like that for schematics.

File: apps/docs-app/docs/integrations/nx/index.md [hunk 1]
```diff
@@ -115,3 +115,27 @@ The client build artifacts are located in the dist folder of your Nx workspace.
 
 In the common apps/libs workspace layout, the `analog-app`'s client artifacts are located in the `dist/apps/analog-app/analog/public` directory.
 The server for the API/SSR build artifacts is located in the `dist/apps/analog-app/analog/server` directory.
+
+### Generators
+
+The Analogjs Nx integration also provides a series of generators that help automate some of the frequent tasks inside an Nx Analog workspace, like generating a **Page**, a starting point for a **Blog** (work in progress), or generating **Posts** (work in progress). To use these generators, the **NxConsole** plugin can be installed or they can be invoked manually using:
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/nx/index.md:121]:
```suggestion
The Analog plugin for Nx also provides a series of generators that help automate some of the frequent tasks inside an Analog project, like generating a **Page**. To use these generators, the **Nx Console** extension can be installed or they can be invoked manually using:
```

File: apps/docs-app/docs/integrations/nx/index.md [hunk 2]
```diff
@@ -115,3 +115,27 @@ The client build artifacts are located in the dist folder of your Nx workspace.
...
+
+### Generate a Page
+
+This command will generate a page inside our `pages` folder with minimal configuration out of the box.
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/nx/index.md:125]:
```suggestion
This command generates a page inside the `pages` folder with minimal configuration out of the box.
```

File: apps/docs-app/docs/integrations/schematics/index.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+sidebar_position: 4
+---
+
+# Angular Schematics
+
+## Overview
+
+Analogjs provides a series of schematics that help automate some of the frequent tasks inside an Analog workspace, like generating a **Page**, a starting point for a **Blog** (work in progress), or generating **Posts** (work in progress). To use these schematics we can use this command:
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/schematics/index.md:9]:
```suggestion
Analog provides a series of schematics that help automate some of the frequent tasks inside an Analog workspace, like generating a **Page**. To use these schematics, use the generate command:
```

File: apps/docs-app/docs/integrations/schematics/index.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+### Generate a Page
+
+This command will generate a page inside our `pages` folder with minimal configuration out of the box.
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/schematics/index.md:13]:
```suggestion
This command generates a page inside our `pages` folder with minimal configuration out of the box.
```

File: apps/docs-app/docs/integrations/schematics/index.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+it also works with the Analog specific filenames, **Note: this names needs to be surrounded by single quotes** ex:
+
+```shell
+ng g @analogjs/platform:page --pathname='(blog)' --project=/
+```
+
+The schematic as well accepts subfolders to structure our project properly.
+
+```shell
+ng g @analogjs/platform:page --pathname='products/[products]' --project=/
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/schematics/index.md:28]:
```suggestion
ng g @analogjs/platform:page --pathname='products/[products]'
```

File: apps/docs-app/docs/integrations/schematics/index.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+This command will generate a page inside our `pages` folder with minimal configuration out of the box.
+
+```shell
+ng g @analogjs/platform:page --pathname=index --project=/
+```
+
+it also works with the Analog specific filenames, **Note: this names needs to be surrounded by single quotes** ex:
+
+```shell
+ng g @analogjs/platform:page --pathname='(blog)' --project=/
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/schematics/index.md:22]:
```suggestion
ng g @analogjs/platform:page --pathname='(blog)'
```

File: apps/docs-app/docs/integrations/schematics/index.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+## Overview
+
+Analogjs provides a series of schematics that help automate some of the frequent tasks inside an Analog workspace, like generating a **Page**, a starting point for a **Blog** (work in progress), or generating **Posts** (work in progress). To use these schematics we can use this command:
+
+### Generate a Page
+
+This command will generate a page inside our `pages` folder with minimal configuration out of the box.
+
+```shell
+ng g @analogjs/platform:page --pathname=index --project=/
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/schematics/index.md:16]:
```suggestion
ng g @analogjs/platform:page --pathname=index
```

File: apps/docs-app/sidebars.js [hunk 8]
```diff
@@ -140,6 +140,11 @@ const sidebars = {
           id: 'integrations/angular-material/index',
           label: 'Angular Material',
         },
+        {
```
Comment (brandonroberts) [inline apps/docs-app/sidebars.js:143]: This should probably be a section under `Core Concepts` named `Code Generation`

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: apps/docs-app/docs/integrations/schematics/index.md (hunk 5 above)
Comment (luishcastroc) [reply apps/docs-app/docs/integrations/schematics/index.md:28]: i tried this without the project and it generates the file in the wrong place the '/'

File: apps/docs-app/docs/integrations/schematics/index.md (hunk 6 above)
Comment (luishcastroc) [reply apps/docs-app/docs/integrations/schematics/index.md]: i tried this without the project and it generates the file in the wrong place the '/'

File: apps/docs-app/docs/integrations/schematics/index.md (hunk 7 above)
Comment (luishcastroc) [reply apps/docs-app/docs/integrations/schematics/index.md]: i tried this without the project and it generates the file in the wrong place the '/'

Comment (brandonroberts) [reply apps/docs-app/docs/integrations/schematics/index.md]: Hmm, we should resolve that then, or have project default to `/`. In a single project workspace the default project should be used.

File: apps/docs-app/docs/features/generation/code-generation.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+Analog supports automated code generation using Nx Generators and Angular Schematics.
+
+## Generate a Page
+
+This command generates a page inside our `pages` folder with minimal configuration out of the box.
+
+<Tabs groupId="project-type">
+  <TabItem value="Nx">
+
+## Generators
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/generation/code-generation.md:15]:
```suggestion
### Generators
```

File: apps/docs-app/docs/features/generation/code-generation.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+```shell
+npx nx generate @analogjs/platform:page --pathname='products/[products]' --project=analog-app
+```
+
+  </TabItem>
+
+  <TabItem label="Schematics" value="schematics">
+
+## Angular Schematics
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/generation/code-generation.md:39]:
```suggestion
### Angular Schematics
```

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**